mysql-test/suite/rocksdb/t/rocksdb_range2.test

```diff
@@ -18,3 +18,63 @@ select count(*) from t1;
 --replace_column 9 # 10 #
 explain select c1 from t1 where c1 > 5 limit 10;
 drop table t1;
+
+--echo #
+--echo # Issue #1052: ha_rocksdb::records_in_range() vastly overestimates the number of rows in certain ranges
+--echo #
+
+CREATE TABLE t1 (
+  part_id smallint(5) unsigned NOT NULL,
+  oid bigint(20) unsigned NOT NULL,
+  tid bigint(20) unsigned NOT NULL,
+  filler char(32),
+  KEY tid (part_id,tid,oid)
+) ENGINE=ROCKSDB;
+
+create table t2(a int primary key);
```

Comment:
Missing engine=rocksdb? Or is it expected and OK here?

Reply:
It seems that `engine=rocksdb` is not possible here:
```
'insert into t3 select A.a + B.a* 10 + C.a * 100 from t2 A, t2 B, t2 C' failed: 1105: Using Gap Lock without full unique key in multi-table or multi-statement transactions is not allowed. You need to either rewrite queries to use all unique key columns in WHERE equal conditions, or rewrite to single-table, single-statement transaction. 
```

Reply:
his is gap lock detection where PS is more restrictive/correct than FB MySQL.  w/o testing engine=rocksdb, the test is useless.  Typical workaround is to change session tx iso mode to read-committed where tx iso mode is not particularly relevant to the feature/function being tested.

Reply:
Fixed